libcudacxx/include/cuda/__mdspan/optimized_accessors.h

```diff
@@ -0,0 +1,184 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA__MDSPAN_OPTIMIZED_ACCESSORS_H
+#define _CUDA__MDSPAN_OPTIMIZED_ACCESSORS_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if _CCCL_STD_VER >= 2014
+
+#  include <cub/thread/thread_load.cuh>
+#  include <cub/thread/thread_store.cuh>
+
+#  include <cuda/std/__cccl/attributes.h>
+#  include <cuda/std/__cccl/compiler.h>
+#  include <cuda/std/__cccl/dialect.h>
+#  include <cuda/std/__cccl/execution_space.h>
```

Comment:
All those are internal to the __cccl_config and are already transitively included through `<cuda/std/detail/__config>`

I aggree that those are special, but I want to avoid pulling in those headers, because there are some ordering concerns there